libs/blocks/nav-marquee/nav-marquee.js

```diff
@@ -0,0 +1,134 @@
+
+function findAnchorTarget(text) {
+  let linkText = text.toLowerCase();
+  if (linkText.charAt(0) === '_') {
+    linkText = linkText.slice(1).replaceAll('_', '-').replaceAll(/[ /|&;$%@"<>()+,.]/g, '');
+    linkText = `#${linkText}`;
+  }
+  return linkText;
+}
+
+let tabNumber = 0;
+function incrementTab() {
+  tabNumber += 1;
+  return tabNumber;
+}
+
+function getItem(title, description, target) {
+  if (!title) return null;
+  const item = createTag('li', { class: 'offer-item' });
+  item.setAttribute('tabNumber', incrementTab(tabNumber));
+  const linkText = createTag('div', { class: 'offer-link-text' });
+  const pageTop = document.querySelector('header')?.offsetHeight ?? 0;
+  const link = createTag('a', { class: 'section-title', href: target, target: '_self' }, title);
+  linkText.append(link);
+  item.addEventListener('click', () => {
+    const isTextSelected = window.getSelection().toString();
+
+    if (!isTextSelected) {
+      link.click();
+    }
+  });
+  link.addEventListener('click', (e) => {
+    const targetPosition = target?.getBoundingClientRect()?.top ?? 0;
+    const offsetPosition = targetPosition + window.pageYOffset - pageTop;
+
+    e.preventDefault();
+    window.scrollTo(0, offsetPosition);
+    target?.setAttribute('tabindex', -1);
+    target?.focus();
+  });
+
+  if (description) linkText.append(createTag('p', { class: 'section-description' }, description));
+  item.append(linkText);
+  item.append(createTag('div', { class: 'offer-arrow' }, DOWN_ARROW_ICON));
+  return item;
+}
+
+function decorateText(el, size) {
+  const headings = el.querySelectorAll('h1, h2, h3, h4, h5, h6');
+  const heading = headings[headings.length - 1];
+  const decorate = (headingEl, headingSize, bodySize, detailSize) => {
+    headingEl.classList.add(`heading-${headingSize}`);
+    const { nextElementSibling } = headingEl;
+    nextElementSibling?.classList.add(`body-${bodySize}`);
+    const { previousElementSibling } = headingEl;
+    if (previousElementSibling) {
+      previousElementSibling.classList.add(
+        previousElementSibling.querySelector('img, .icon') ? 'icon-area' : `detail-${detailSize}`
+      );
+      previousElementSibling.previousElementSibling?.classList.add('icon-area');
+    }
+  };
+  const headingSize = size === 'large' ? 'xxl' : 'xl';
+  const bodySize = size === 'large' ? 'xl' : 'm';
+  const detailSize = size === 'large' ? 'l' : 'm';
+  decorate(heading, headingSize, bodySize, detailSize);
+}
+
+export default function init(el) {
+  const size = getBlockSize(el);
+  const children = Array.from(el.querySelectorAll(':scope > div'));
+  // const tone = (el.classList.contains('dark')) ? 'dark' : 'light';
+  // el.classList.add(tone);
+  const marquee = createTag('div', { class: 'marquee' }, children.shift());
```

Comment:
I was wondering if your intentionally adopting and using .marquee styles? 
Otherwise we should probably name this different so there isn't a naming collision.

Reply:
There appears to be an un necessary amount of createTag() usage going on. We should try to look first at the natural dom structure and see if we can work w/ that where possible. I have this branch pulled down locally and am testing a few things if you wanna sync up in a few @trixyfox